### PR TITLE
docs: update userflow with option to download and upload table metadata

### LINKED ISSUE #101: [feature-request]: Add option to download and upload table metadata to user flow
### What feature would you like to request?

Based on the user stories #100 and #99, it makes sense to add these features to the user flow post and diagrams

### Optional: Suggest a solution

Update the following files: 
- user_flow.qmd 
- images/user_flow.puml 
- images/user_flow.png

as well as the subpart user flow diagrams (pumls+pngs) where the changes appear

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds feature of uploading and downloading table metadata (as described in #99 and #100) in the user flow diagrams and user flow post.
- NB: While doing this, I realised that the details of the user flow might need editing later on. Therefore, I added table metadata upload and download to the user flow as well as made other updates to the diagrams. However, I did not finish the work of adding the changes to the user flow post (`user-flow.png`). I don't think we should spend too much time on this at this point in time. 

## Related Issues

Closes #101

See also Issues #99 and #100.

## Reviewer Focus

Reviewers should focus on the newly included steps in the diagrams and whether it makes sense to place them there in the user flow. This should just be a *brief* review. No need to go into too much detail. This might change later on anyways.

## Checklist

- [X] Add upload table metadata to user flow diagram (and update png)
- [X] Add upload of table metadata to user flow post
- [X] Update relevant subpart user flow diagram with added upload steps
- [X] Add download of table metadata to user flow diagram (and update png)
- [ ] Add download of table metadata to user flow post
- [X] Update relevant subpart user flow diagram with added download steps

For general documentation:
- [ ] Spell-check
    - [ ] UK
- [ ] Did the page(s) preview correctly on your machine without breaking
- [X] New category words (keywords) (if any) added to the code snippet file

### CODE REVIEW

**Review by philter87 — DISMISSED** [review]

File: docs/design/images/user-flow-table-created-continue-to-upload.png
Comment (philter87) [inline docs/design/images/user-flow-table-created-continue-to-upload.png:1]: Why would the user download the metadata?

**Review by philter87 — DISMISSED** [review]

File: docs/design/images/user-flow-table-created-continue-to-upload.png
Comment (signekb) [reply docs/design/images/user-flow-table-created-continue-to-upload.png:1]: See #99

File: docs/design/user-flow.qmd [hunk 1]
```diff
@@ -8,48 +8,69 @@ This document describes the user flow of Sprout. The user flow is split into fiv
...
-| `data_resource` | `table_1` | `first_name` | User's first name | `varchar` | 0 | NULL |
-| `data_resource` | `table_1` | `email` | Unique ID identifying the user | `int` | 0 | NULL |
-| `data_resource` | `table_1` | `address` | first address line of user's address | `varchar` | 1 | address line 1 |
+A [Data Management Administrator](users.qmd#data-management-administrators) only needs to write the table name and description. The remaining information is extracted by Sprout behind the scenes. 
+
+Whenever a new table metadata is added, Sprout will check whether a table with the exact same name already exists. It will not be possible to create a table with a name that already exists, and the user will be asked to write a different name for the table.
+
+![Diagram of the flow that occurs when table metadata is created.](images/user-flow-table-metadata-creation.png)
+
+After a table's metadata has been created, the next step is to define the columsn within the table. Column metadata includes:
```
Comment (lwjohnst86) [inline docs/design/user-flow.qmd:39]:
```suggestion
After a table's metadata has been created, the next step is to define the columns within the table. Column metadata includes:
```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Small typo, otherwise it looks great! I'll apply the suggestion, approve, then merge.

**Review by lwjohnst86 — DISMISSED** [review]

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**